### PR TITLE
CGMES conversion node container mapping test

### PR DESCRIPTION
**Pull request by rcourtier**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

New unit tests

**What is the current behavior?**

Node container mapping tests are using modified conformity networks

**What is the new behavior (if this is a feature change)?**

Node container mapping tests are using simple networks

**Does this PR introduce a breaking change or deprecate an API?**
- [X] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [X] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

`SubstationIdMapping` was renamed to `NodeContainerMapping`.  
As a consequence, method `substationIdMapping()` of `com.powsybl.cgmes.conversion.Context` was renamed to `nodeContainerMapping()`.

The following methods of `CgmesConformity1ModifiedCatalog`, were removed:
- `miniNodeBreakerCimLine()`
- `miniNodeBreakerSwitchBetweenVoltageLevelsOpen()`
- `miniNodeBreakerJoinVoltageLevelSwitch()`
- `miniNodeBreakerJoinVoltageLevelTx()`
- `miniNodeBreakerSubstationNode()`

They were used for unit testing, but they relied on big CGMES files ; this slowed the tests execution. If you still need to use them, you can retrieve the methods at [this URL](https://github.com/powsybl/powsybl-core/blob/5e36b26d8736f03a4a6edd6f42a5e6f4c52d7972/cgmes/cgmes-conformity/src/main/java/com/powsybl/cgmes/conformity/CgmesConformity1ModifiedCatalog.java) and the corresponding CGMES files in [this directory](https://github.com/powsybl/powsybl-core/tree/5e36b26d8736f03a4a6edd6f42a5e6f4c52d7972/cgmes/cgmes-conformity/src/main/resources/conformity-modified/cas-1.1.3-data-4.0.3/MiniGrid/NodeBreaker). 

**Other information**:

### CODE REVIEW

File: cgmes/cgmes-conversion/src/test/resources/issues/node-containers/line_with_t-junction.xml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    <cim:ACLineSegment.g0ch>0</cim:ACLineSegment.g0ch>
+    <cim:ACLineSegment.gch>0</cim:ACLineSegment.gch>
+    <cim:IdentifiedObject.name>Line 1A</cim:IdentifiedObject.name>
+    <cim:ACLineSegment.r>0.1</cim:ACLineSegment.r>
+    <cim:ACLineSegment.r0>0.1</cim:ACLineSegment.r0>
+    <cim:ACLineSegment.shortCircuitEndTemperature>75</cim:ACLineSegment.shortCircuitEndTemperature>
+    <cim:ACLineSegment.x>1</cim:ACLineSegment.x>
+    <cim:ACLineSegment.x0>1</cim:ACLineSegment.x0>
+    <cim:ConductingEquipment.BaseVoltage rdf:resource="#_BV"/>
+    <cim:Equipment.EquipmentContainer rdf:resource="#_LN_A"/>
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/test/resources/issues/node-containers/line_with_t-junction.xml:77]: We could remove the `Equipment.EquipmentContainer` in one or more line segments to check that we do not depend on this optional association.

**Review by zamarrenolm — COMMENTED** [review]
Thanks for the simplification of the unit tests; removing the dependencies from ENTSO-E test configurations is great: it will speed up the tests and reduce the size of the required resources. Also, the cases are easier to follow, and they are very well documented. 

There are some minor changes proposed. 

Only one previously tested configuration is lost with this change, but I think it is not required to recover it. It was created from ENTSO-E configuration mini node/breaker, where both end nodes of a line segment were contained in a line container. (`CgmesConformity1ModifiedCatalog::miniNodeBreakerCimLine`). This and other corner cases (like chains of T-junctions) are also supported with current implementation and have been verified with real-world examples.

File: cgmes/cgmes-conversion/src/test/resources/issues/node-containers/line_with_t-junction.xml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+</cim:ACLineSegment>
+<cim:Terminal rdf:ID="_T_3A_1">
+    <cim:IdentifiedObject.name>Terminal 3A 1</cim:IdentifiedObject.name>
+    <cim:ACDCTerminal.sequenceNumber>1</cim:ACDCTerminal.sequenceNumber>
+    <cim:Terminal.ConductingEquipment rdf:resource="#_ACL_3A"/>
+    <cim:Terminal.ConnectivityNode rdf:resource="#_CN_1"/>
```
[the remark this answers is not in the document]
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/test/resources/issues/node-containers/line_with_t-junction.xml]: You are absolutely right. Thanks for noticing!
Fixed in latest commit.

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/ConversionUtil.java [hunk 3]
```diff
@@ -84,4 +85,13 @@ public static boolean xmlContains(InputStream is, String clazz, String ns, Strin
         }
         return false;
     }
+
+    public static Network readCgmesFiles(String dir, String... files) {
+        return readCgmesFiles(new Properties(), dir, files);
+    }
+
+    public static Network readCgmesFiles(Properties properties, String dir, String... files) {
```
Comment (zamarrenolm) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/ConversionUtil.java:93]: Maybe consider `readCgmesResources` to avoid confusions with potential reading from file system?

File: cgmes/cgmes-conversion/src/test/resources/issues/node-containers/line_with_t-junction.xml (hunk 1 above)
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/test/resources/issues/node-containers/line_with_t-junction.xml:77]: Done. I have merged it with the same test in LineContainersTest.

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/ConversionUtil.java (hunk 3 above)
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/ConversionUtil.java]: Done.

**Review by zamarrenolm — APPROVED** [review]

**Comment by rcourtier** [comment]
I have added a test with an ACLineSegment having both its nodes in the Line container, and checked the import with two alternatives for the fictitious VoltageLevel creation.

**Comment by zamarrenolm** [comment]
> I have added a test with an ACLineSegment having both its nodes in the Line container, and checked the import with two alternatives for the fictitious VoltageLevel creation.

Perfect. I've verified the definition of the case, it mimics the configuration that was used in the removed `CgmesConformity1ModifiedCatalog::miniNodeBreakerCimLine`, so we are not losing any unit test, and you have added a few more. Great.
